public/app/features/profile/ChangePasswordForm.tsx

```diff
@@ -16,10 +17,86 @@ export interface Props {
   onChangePassword: (payload: ChangePasswordFields) => void;
 }
 
+interface StrongPasswordValidation {
+  message: string;
+  validation: (value: string) => boolean;
+}
+
+const strongPasswordValidations: StrongPasswordValidation[] = [
+  {
+    message: 'At least 12 characters',
+    validation: (value: string) => value.length >= 12,
+  },
+  {
+    message: 'One uppercase letter',
+    validation: (value: string) => /[A-Z]+/.test(value),
+  },
+  {
+    message: 'One lowercase letter',
+    validation: (value: string) => /[a-z]+/.test(value),
+  },
+  {
+    message: 'One number',
+    validation: (value: string) => /[0-9]+/.test(value),
+  },
+  {
+    message: 'One symbol',
+    validation: (value: string) => /[\W]/.test(value),
+  },
+];
+
 export const ChangePasswordForm = ({ user, onChangePassword, isSaving }: Props) => {
+  const [displayValidationLabels, setDisplayValidationLabels] = useState(false);
+  const [pristine, setPristine] = useState(true);
+  const [newPassword, setNewPassword] = useState('');
+
   const { disableLoginForm } = config;
+  const { basicAuthStrongPasswordPolicy } = config.auth;
   const authSource = user.authLabels?.length && user.authLabels[0];
 
+  const validationLabel = (index: number, message: string, validation: () => {}) => {
+    if (!basicAuthStrongPasswordPolicy) {
+      return null;
+    }
+    const result = newPassword.length > 0 && validation();
```

Comment:
I reworked this function to accept a StrongPasswordValidation interface instead. I didn't change the validation to its result because the password length was evaluated with the validation. Let me know if you feel strongly about it.

Reply:
No strong preferences, it's jus that imo it's easier to read the code.